src/s1ap/handlers/s1ap_msg_delegator.c

```diff
@@ -318,7 +318,24 @@ parse_IEs(char *msg, struct proto_IE *proto_ies, unsigned short proc_code)
 		ie->IE_type = IE_type;
 		msg +=2;//next to ie type
 		msg +=1;//next to criticality
-		memcpy(&IE_data_len, msg, sizeof(char));
+        /*parse length according to PER rules*/
```

Comment:
unsigned char val = msg[0] & 0xc0;
        val = (val >> 6);
        if(val == 2)
        {
            log_msg(LOG_INFO, "length more than 128\n");
            unsigned short first = (unsigned char)msg[0] & 0x3f;
            msg += 1;
            unsigned short second = (unsigned char)msg[0];
            unsigned short ie_len = second | (first << 8);
            IE_data_len = ie_len;
        }
        else
        {
            log_msg(LOG_INFO, "length less than 128\n");
            memcpy(&IE_data_len, msg, sizeof(char));
        }

Reply:
Could you please also add a \n here since you are at it?
	log_msg(LOG_INFO, "--------------------- %d --------------", msg[3]);